src/context.c

```diff
@@ -1645,8 +1645,7 @@ getdns_context_set_namespaces(getdns_context *context,
 	size_t i;
 	getdns_return_t r = GETDNS_RETURN_GOOD;
 
-	if (!context)
-		return GETDNS_RETURN_INVALID_PARAMETER;
+	RETURN_IF_NULL(context, GETDNS_RETURN_INVALID_PARAMETER);
```

Comment:
Op 18-05-16 om 11:12 schreef saradickinson:
> In src/context.c
> <https://github.com/getdnsapi/getdns/pull/172#discussion_r63669797>:
> 
>> @@ -1645,8 +1645,7 @@ getdns_context_set_namespaces(getdns_context *context,
>>  	size_t i;
>>  	getdns_return_t r = GETDNS_RETURN_GOOD;
>>  
>> -	if (!context)
>> -		return GETDNS_RETURN_INVALID_PARAMETER;
>> +	RETURN_IF_NULL(context, GETDNS_RETURN_INVALID_PARAMETER);
> 
> I was just trying to be more consistent with the other existing
> functions - the macro was already there and was used in many other set()
> functions. I don't mind which one we use, but I like code consistency.

I would like to plead for tolerance in this respect.  I think it is more
important that commits contain the changes related to the purpose of the
commit.

> I also have a question about the return codes. In inspecting all the
> existing setters, the GETDNS_RETURN_INVALID_PARAMETER is used if the
> context is NULL and as far as I can see
> GETDNS_RETURN_CONTEXT_UPDATE_FAIL is used for all other cases, even when
> a passed parameter is invalid (null pointer to list, invalid enum). This
> seems a bit lacking, but I don't know if this was a design decision way
> back?

I believe spec mentions only GETDNS_RETURN_CONTEXT_UPDATE_FAIL.  But I
think it is way preferable to return the most specific error message
whenever possible.

> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/getdnsapi/getdns/pull/172/files/5f225d6be3792771a4ec2a476e5eb141ebbed936..98d636b99d8be903c316e1ca4e279f14b5b6e390#r63669797>
>